src/Forms/SearchableDropdownTrait.php

```diff
@@ -435,6 +435,18 @@ public function validate($validator): bool
         return $this->extendValidationResult(true, $validator);
     }
 
+    public function dataValue()
+    {
+        $name = $this->getName();
+        $ids = $this->getValueArray();
+        // This is how we distinguish between single and multi fields in saveInto()
+        // so use the same logic here.
+        if (substr($name, -2) === 'ID') {
+            return $ids[0] ?? 0;
```

Comment:
Are you completely use we should have ` ?? 0` - looking at `SearchableDropdownTraitTest::provideGetValueArray()` it kind of seems like the expected value for "no value" isn't `0`, it's perhaps closer to blank string, perhaps null?

We can't assume that this field will be backed by a DBInt, though if it is then a blank string would get converted to a 0 anyway?

Reply:
Regardless of what DBField this is backed by, this field is exclusively and explicitly intended for dealing with record IDs. It's accepted that `0` is the "no record" value for IDs basically everywhere in framework.